pandas/core/indexes/datetimes.py

```diff
@@ -954,22 +965,19 @@ def _add_offset(self, offset):
 
     def _add_offset_array(self, other):
         # Array/Index of DateOffset objects
-        if isinstance(other, ABCSeries):
-            return NotImplemented
-        elif len(other) == 1:
+        if len(other) == 1:
             return self + other[0]
         else:
             warnings.warn("Adding/subtracting array of DateOffsets to "
                           "{} not vectorized".format(type(self)),
                           PerformanceWarning)
             return self.astype('O') + np.array(other)
+            # TODO: pass freq='infer' like we do in _sub_offset_array?
```

Comment:
what is the purpose of these comments (esp here, after the return)?

Reply:
a) to make sure it gets seen during review, b) as a note to myself or the next pass